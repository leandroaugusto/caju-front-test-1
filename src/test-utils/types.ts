export interface ITestUtils {
  children?: React.ReactNode
}
