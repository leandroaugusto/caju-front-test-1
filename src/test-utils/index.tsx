import { ReactElement } from "react";
import { ThemeProvider } from "styled-components";
import { render, RenderOptions, RenderResult } from "@testing-library/react";

import { ITestUtils } from "./types";

import theme from "~/theme";

const AllTheProviders: React.FC = ({ children }: ITestUtils) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
): RenderResult =>
  render(ui, {
    wrapper: AllTheProviders,
    ...options,
  });

export * from "@testing-library/react";

export { customRender };
