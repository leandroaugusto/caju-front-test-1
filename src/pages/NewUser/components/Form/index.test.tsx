import { FieldError } from "react-hook-form";

import { Form } from ".";

import { customRender } from "~/test-utils";

const mockOnSubmit = vi.fn();

const defaultProps = {
  onSubmit: mockOnSubmit,
  errors: {},
  register: vi.fn(),
  registerWithMask: vi.fn(),
};

describe("Form", () => {
  it("Should render Form", () => {
    const { container } = customRender(<Form {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });

  it("Should render Form with errors", () => {
    const errorMessage = "Required";
    const errors = {
      employeeName: { message: errorMessage } as FieldError,
      email: { message: errorMessage } as FieldError,
      cpf: { message: errorMessage } as FieldError,
      admissionDate: { message: errorMessage } as FieldError,
    };

    const { container } = customRender(
      <Form {...defaultProps} errors={errors} />
    );

    expect(container).toMatchSnapshot();
  });

  it("Should call onSubmit function", () => {
    const { getByText } = customRender(<Form {...defaultProps} />);
    const button = getByText("Cadastrar");

    button.click();

    expect(mockOnSubmit).toHaveBeenCalledTimes(1);
  });
});
