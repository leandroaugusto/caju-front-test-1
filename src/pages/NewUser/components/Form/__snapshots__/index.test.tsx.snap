// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Form > Should render Form 1`] = `
<div>
  <form
    class="sc-gnpbhQ jfSDEr"
  >
    <fieldset
      class="sc-braxZu iUDKFC"
    >
      <label
        class="sc-gJhJTp kIXmzW"
        for="employeeName"
      >
        Nome
      </label>
      <input
        class="sc-elDIKY jqaaIW"
        id="employeeName"
        name="employeeName"
        placeholder="Nome"
        type="text"
      />
      <span
        class="sc-fQpRED dUfiZf"
      />
    </fieldset>
    <fieldset
      class="sc-braxZu iUDKFC"
    >
      <label
        class="sc-gJhJTp kIXmzW"
        for="email"
      >
        Email
      </label>
      <input
        class="sc-elDIKY jqaaIW"
        id="email"
        name="email"
        placeholder="Email"
        type="email"
      />
      <span
        class="sc-fQpRED dUfiZf"
      />
    </fieldset>
    <fieldset
      class="sc-braxZu iUDKFC"
    >
      <label
        class="sc-gJhJTp kIXmzW"
        for="cpf"
      >
        CPF
      </label>
      <input
        class="sc-elDIKY jqaaIW"
        id="cpf"
        name="cpf"
        placeholder="CPF"
        type="tel"
      />
      <span
        class="sc-fQpRED dUfiZf"
      />
    </fieldset>
    <fieldset
      class="sc-braxZu iUDKFC"
    >
      <label
        class="sc-gJhJTp kIXmzW"
        for="admissionDate"
      >
        Data de admiss達o
      </label>
      <input
        class="sc-elDIKY jqaaIW"
        id="admissionDate"
        name="admissionDate"
        placeholder="Data de admiss達o"
        type="date"
      />
      <span
        class="sc-fQpRED dUfiZf"
      />
    </fieldset>
    <button
      class="sc-dsLQwm bIItHX"
      type="submit"
    >
      Cadastrar
    </button>
  </form>
</div>
`;

exports[`Form > Should render Form with errors 1`] = `
<div>
  <form
    class="sc-gnpbhQ jfSDEr"
  >
    <fieldset
      class="sc-braxZu iUDKFC"
    >
      <label
        class="sc-gJhJTp HOZvf"
        for="employeeName"
      >
        Nome
      </label>
      <input
        class="sc-elDIKY fPchqO"
        id="employeeName"
        name="employeeName"
        placeholder="Nome"
        type="text"
      />
      <span
        class="sc-fQpRED dUfiZf"
      >
        Required
      </span>
    </fieldset>
    <fieldset
      class="sc-braxZu iUDKFC"
    >
      <label
        class="sc-gJhJTp HOZvf"
        for="email"
      >
        Email
      </label>
      <input
        class="sc-elDIKY fPchqO"
        id="email"
        name="email"
        placeholder="Email"
        type="email"
      />
      <span
        class="sc-fQpRED dUfiZf"
      >
        Required
      </span>
    </fieldset>
    <fieldset
      class="sc-braxZu iUDKFC"
    >
      <label
        class="sc-gJhJTp HOZvf"
        for="cpf"
      >
        CPF
      </label>
      <input
        class="sc-elDIKY fPchqO"
        id="cpf"
        name="cpf"
        placeholder="CPF"
        type="tel"
      />
      <span
        class="sc-fQpRED dUfiZf"
      >
        Required
      </span>
    </fieldset>
    <fieldset
      class="sc-braxZu iUDKFC"
    >
      <label
        class="sc-gJhJTp HOZvf"
        for="admissionDate"
      >
        Data de admiss達o
      </label>
      <input
        class="sc-elDIKY fPchqO"
        id="admissionDate"
        name="admissionDate"
        placeholder="Data de admiss達o"
        type="date"
      />
      <span
        class="sc-fQpRED dUfiZf"
      >
        Required
      </span>
    </fieldset>
    <button
      class="sc-dsLQwm bIItHX"
      type="submit"
    >
      Cadastrar
    </button>
  </form>
</div>
`;
